src/Riok.Mapperly/Descriptors/Enumerables/EnsureCapacity/EnsureCapacity.cs

```diff
@@ -0,0 +1,21 @@
+﻿using Microsoft.CodeAnalysis.CSharp;
+using Microsoft.CodeAnalysis.CSharp.Syntax;
+using Riok.Mapperly.Descriptors.Mappings;
+using static Microsoft.CodeAnalysis.CSharp.SyntaxFactory;
+using static Riok.Mapperly.Emit.SyntaxFactoryHelper;
+
+
+namespace Riok.Mapperly.Descriptors.Enumerables.EnsureCapacity;
+
+public abstract class EnsureCapacity
+{
+    private const string EnsureCapacityName = "EnsureCapacity";
+
+    public abstract StatementSyntax BuildEnsureCapacityStatement(TypeMappingBuildContext ctx, ExpressionSyntax target);
```

Comment:
I'd simplify this method name. There is only one build method in this class which is exposed and since the class is named `EnsureCapacity` it should be clear that it is an ensure capacity statement. IMO `Build` should be simple and understandable.